test/e2e/operator_test.go

```diff
@@ -455,118 +280,76 @@ u3YLAbyW/lHhOCiZu2iAI8AbmXem9lW6Tr7p/97s0w==
 //
 // See also: https://github.com/kubernetes/kubernetes/pull/75210
 func TestIngressControllerScale(t *testing.T) {
-	cl, ns, err := getClient()
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	name := types.NamespacedName{Namespace: ns, Name: manifests.DefaultIngressControllerName}
-	err = waitForIngressControllerCondition(cl, 15*time.Second, name, defaultAvailableConditions...)
-	if err != nil {
-		t.Errorf("failed to observe expected conditions: %v", err)
-	}
-
 	ic := &operatorv1.IngressController{}
-	if err := cl.Get(context.TODO(), name, ic); err != nil {
-		t.Fatalf("failed to get ingresscontroller %s: %v", name, err)
+	if err := kclient.Get(context.TODO(), defaultName, ic); err != nil {
+		t.Fatalf("failed to get default ingresscontroller: %v", err)
+	}
+	if err := waitForIngressControllerCondition(kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
 	}
 
-	deploymentName := controller.RouterDeploymentName(ic)
+	// TODO: desired replicas isn't actually persisted in our API...
 	deployment := &appsv1.Deployment{}
-	if err := cl.Get(context.TODO(), deploymentName, deployment); err != nil {
-		t.Fatalf("failed to get deployment %s: %v", deploymentName, err)
+	if err := kclient.Get(context.TODO(), controller.RouterDeploymentName(ic), deployment); err != nil {
+		t.Fatalf("failed to get default ingresscontroller deployment: %v", err)
 	}
+	originalReplicas := deployment.Spec.Replicas
+	newReplicas := *originalReplicas + 1
 
-	originalReplicas := *deployment.Spec.Replicas
-	newReplicas := originalReplicas + 1
-
-	if err := cl.Get(context.TODO(), name, ic); err != nil {
-		t.Fatalf("failed to get ingresscontroller %s: %v", name, err)
-	}
 	ic.Spec.Replicas = &newReplicas
-	if err := cl.Update(context.TODO(), ic); err != nil {
+	if err := kclient.Update(context.TODO(), ic); err != nil {
 		t.Fatalf("failed to update ingresscontroller: %v", err)
 	}
 
 	// Wait for the deployment scale up to be observed.
-	err = waitForAvailableReplicas(cl, name, 2*time.Minute, newReplicas)
-	if err != nil {
-		t.Fatalf("failed waiting deployment %s to scale to %d: %v", deploymentName, newReplicas, err)
+	if err := waitForAvailableReplicas(kclient, ic, 2*time.Minute, newReplicas); err != nil {
+		t.Fatalf("failed waiting deployment %s to scale to %d: %v", defaultName, newReplicas, err)
 	}
 
 	// Ensure the ingresscontroller remains available
-	err = waitForIngressControllerCondition(cl, 2*time.Minute, name, defaultAvailableConditions...)
-	if err != nil {
-		t.Errorf("failed to observe expected conditions: %v", err)
+	if err := waitForIngressControllerCondition(kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
```

Comment:
Was a 2-minute timeout insufficient?  What do we need to wait for at this point anyway?  We already have waited for up to 5 minutes for DNS and the load balancer at the beginning of the test, and then for up to 2 minutes for the replicas to be available after scale-up.

Reply:
Just an oversight, reverted to 2 minutes